lib/nostrum/struct/sticker.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Nostrum.Struct.Sticker do
+  @moduledoc """
+  A `Nostrum.Struct.Sticker` represents a sticker that can be sent inside a
+  `Nostrum.Struct.Message` and created/modified with the relevant API methods
+  """
+
+  alias Nostrum.Struct.{Guild, User}
+  alias Nostrum.{Constants, Snowflake, Util}
+
+  defstruct [
+    :id,
+    :pack_id,
+    :name,
+    :description,
+    :tags,
+    :type,
+    :format_type,
+    :available,
+    :guild_id,
+    :user,
+    :sort_value
+  ]
+
+  @typedoc """
+  ID of the sticker
+  """
+  @type id :: Snowflake.t()
+
+  @typedoc """
+  ID of the pack the sticker is from
+  """
+  @type pack_id :: Snowflake.t()
+
+  @typedoc """
+  Name of the sticker
+  """
+  @type name :: String.t() | nil
+
+  @typedoc """
+  Description of the sticker
+  """
+  @type description :: String.t() | nil
+
+  @typedoc """
+  Tags used by the Discord client to auto-complete a sticker.
+
+  For default sticker packs, this is a comma-separated list. For guild stickers,
+  this is the name of the unicode emoji associated by the sticker creator with
+  the sticker.
+
+  This is technically a free-text field so consistency in formatting is not guaranteed.
+  """
+  @type tags :: String.t()
+
+  @typedoc """
+  Type of the sticker as an integer enum.
+  """
+  @type type :: :standard | :guild
+
+  @typedoc """
+  Format of the sticker. This field is used to determine the return URL in `cdn_url/1`.
+  """
+  @type format_type :: :png | :apng | :lottie | :gif
+
+  @typedoc """
+  Whether this guild sticker can be used, may be false due to loss of Server Boosts
+  """
+  @type available :: boolean
+
+  @typedoc """
+  Id of the guild that owns this sticker
+  """
+  @type guild_id :: Guild.id() | nil
+
+  @typedoc """
+  User that uploaded the guild sticker
+  """
+  @type user :: User.t() | nil
+
+  @typedoc """
+  The sticker's sort order within its pack
+  """
+  @type sort_value :: integer() | nil
```

Comment:
When is this nil? How can users depend on it? Which direction should it be rendered in?

Reply:
Only filled for pack information from Discord. I think the use case for needing this field is niche enough that users should know the information. No custom stickers can have this field set.

I believe it's integer ascending but there is no authoritative source on this.